python/ray/_private/worker.py

```diff
@@ -1399,16 +1402,26 @@ def init(
                 "job_config. Please ensure no runtime_env is used in driver "
                 "script's ray.init() when using job submission API."
             )
-        # Set runtime_env in job_config if passed as env variable, such as
-        # ray job submission with driver script executed in subprocess
-        job_config_json = json.loads(os.environ.get(RAY_JOB_CONFIG_JSON_ENV_VAR))
-        job_config = ray.job_config.JobConfig.from_json(job_config_json)
+        injected_job_config_json = json.loads(
+            os.environ.get(RAY_JOB_CONFIG_JSON_ENV_VAR)
```

Comment:
at high level, seems we should always allow user's job_config to overwrite RAY_JOB_CONFIG_JSON_ENV_VAR, would this work?

cc @architkulkarni

Reply:
@scv119 we actually discussed this with @architkulkarni:

 - `RAY_JOB_CONFIG_JSON_ENV_VAR` injects RuntimeEnv from the Job submission API 
 - In case someone is using Job submission API we want to direct them to set `runtime_env` in submit_job API to make sure they are not specifying _both_ (which is unspecified behavior)

Therefore we always prefer `RAY_JOB_CONFIG_JSON_ENV_VAR` if provided

Reply:
> In case someone is using Job submission API we want to direct them to set runtime_env in submit_job API to make sure they are not specifying both (which is unspecified behavior)

in this case should we log some warning message if undefined behavior happens.

Reply:
We do log this warning at L1396